lib/tests/config_node_test.cc

```diff
@@ -92,4 +94,240 @@ TEST_CASE("simple values", "[config-node]") {
+    REQUIRE("bar : 15" == node->render());
+    auto new_node = node->set_value_on_path("foo", value);
+    string final_text = "bar : 15, foo : " + value->render();
+    REQUIRE(final_text == new_node->render());
+}
+
+TEST_CASE("creates non-existent paths", "[config-node]") {
+    nonexistent_path_test(int_node(10));
+    nonexistent_path_test(make_shared<config_node_object>(
+            shared_node_list { open_brace_node(),
+                               node_key_value_pair(node_key("foo"), double_node(3.14)),
+                               close_brace_node() }));
+    nonexistent_path_test(make_shared<config_node_array>(shared_node_list {
+            open_brace_node(),
+            int_node(15),
+            close_brace_node()
+    }));
+}
+
+TEST_CASE("replace nested nodes") {
+    string orig_text = string("foo : bar\nbaz : {\n\t\"abc.def\" : 123\n\t") +
+                       string("//this is a comment about the below setting") +
+                       string("\n\n\tabc : {\n\t\tdef : \"this is a string\"") +
+                       string("\n\t\tghi : ${\"a.b\"}\n\t}\n}\nbaz.abc.ghi ") +
+                       string(": 52\nbaz.abc.ghi : 53\n}");
+    auto lowest_level_map = make_shared<config_node_object>(shared_node_list {
+          open_brace_node(),
+          line_node(6),
+          whitespace_node("\t\t"),
+          node_key_value_pair(node_key("def"), string_node("\"this is a string\"")),
+          line_node(7),
+          whitespace_node("\t\t"),
+          node_key_value_pair(node_key("ghi"), substitution_node(string_token("\"a.b\""), false)),
+          line_node(8),
+          whitespace_node("\t"),
+          close_brace_node()
+    });
+    auto higher_level_map = make_shared<config_node_object>(shared_node_list {
+            open_brace_node(),
+            line_node(2),
+            whitespace_node("\t"),
+            node_key_value_pair(node_key("\"abc.def\""), int_node(123)),
+            line_node(3),
+            whitespace_node("\t"),
+            double_slash_comment_node("this is a comment about the below setting"),
+            line_node(4),
+            line_node(5),
+            whitespace_node("\t"),
+            node_key_value_pair(node_key("abc"), lowest_level_map),
+            line_node(9),
+            close_brace_node()
+    });
+    auto original_node = make_shared<config_node_object>(shared_node_list {
+            node_key_value_pair(node_key("foo"), unquoted_text_node("bar")),
+            line_node(1),
+            node_key_value_pair(node_key("baz"), higher_level_map),
+            line_node(10),
+            node_key_value_pair(node_key("baz.abc.ghi"), int_node(52)),
+            line_node(11),
+            node_key_value_pair(node_key("baz.abc.ghi"), int_node(53)),
+            line_node(12),
+            close_brace_node()
+    });
+    REQUIRE(orig_text == original_node->render());
+
+    string final_text = string("foo : bar\nbaz : {\n\t\"abc.def\" : true\n\t") +
```

Comment:
Instead of concatenating strings, you can concat string literals, as in
```
"foo : bar\n" "baz : "
"{\n\t\"..."
```
This is C-style string literal concatenation, and can span multiple lines.